packages/validator/src/services/attestation.ts

```diff
@@ -276,4 +291,95 @@ export class AttestationService {
+      this.api.validator
+        .submitBeaconCommitteeSelections(partialSelections)
+        .catch((e) => this.logger.error("Error on submitBeaconCommitteeSelections", {slot}, e)),
+      // Exit attestation aggregation flow if there is no response after 1/3 of slot as
+      // beacon node would likely not have enough time to prepare an aggregate attestation.
+      // Note that the aggregations flow is not explicitly exited but rather will be skipped
+      // due to the fact that calculation of `is_aggregator` in duties service is not done
+      // and selectionProof is set to null, meaning no validator will be considered an aggregator.
+      sleep(this.clock.msToSlot(slot + 1 / 3), signal),
+    ]);
+
+    if (!res) {
+      throw new Error("submitBeaconCommitteeSelections did not resolve after 1/3 of slot");
+    }
+    ApiError.assert(res);
+
+    const combinedSelections = res.response.data;
+    this.logger.debug("Received combined beacon committee selection proofs", {slot, count: combinedSelections.length});
+
+    const beaconCommitteeSubscriptions: routes.validator.BeaconCommitteeSubscription[] = [];
+
+    for (const dutyAndProof of duties) {
+      const {validatorIndex, committeeIndex, committeeLength, committeesAtSlot} = dutyAndProof.duty;
+      const logCtxValidator = {slot, index: committeeIndex, validatorIndex};
+
+      const combinedSelection = combinedSelections.find((s) => s.validatorIndex === validatorIndex && s.slot === slot);
+
+      if (!combinedSelection) {
+        this.logger.warn("Did not receive combined beacon committee selection proof", logCtxValidator);
+        continue;
+      }
+
+      const isAggregator = isAggregatorFromCommitteeLength(committeeLength, combinedSelection.selectionProof);
+
+      if (isAggregator) {
+        // Update selection proof by mutating duty object
+        dutyAndProof.selectionProof = combinedSelection.selectionProof;
+
+        // Only push subnet subscriptions with `isAggregator=true` as all validators
+        // with duties for slot are already subscribed to subnets with `isAggregator=false`.
+        beaconCommitteeSubscriptions.push({
+          validatorIndex,
+          committeesAtSlot,
+          committeeIndex,
+          slot,
+          isAggregator,
+        });
+        this.logger.debug("Resubscribing validator as aggregator on beacon committee subnet", logCtxValidator);
+      }
+    }
+
+    // If there are any subscriptions with aggregators, push them out to the beacon node.
+    if (beaconCommitteeSubscriptions.length > 0) {
+      try {
+        ApiError.assert(await this.api.validator.prepareBeaconCommitteeSubnet(beaconCommitteeSubscriptions));
+        this.logger.debug("Resubscribed validators as aggregators on beacon committee subnets", {
+          slot,
+          count: beaconCommitteeSubscriptions.length,
+        });
+      } catch (e) {
+        this.logger.error("Failed to resubscribe to beacon committee subnets", {slot}, e as Error);
```

Comment:
to be considered, throwing vs catching + logging the error here, if thrown would be logged by top-level catch attached to `runDistributedAggregationSelectionTasks`

Reply:
better to throw error here and log in top-level catch as it will add more context `Error on attestation aggregation selection...` to error message